### DISCUSSION TITLE
Question: Is it possible to split .exe creation and service creation

### DISCUSSION THREAD

**Discussion by raikm** [coreybutler/node-windows#349] [Q&A]
Is it possible to use this library to create a ```.exe``` file for shipping to customers where the service gets onside created? So I would like to avoid shipping the whole node project and would rather send the ```.exe``` file to the customer. 

Thanks in advance!

**Comment by coreybutler** [comment]
This library provides a service runner, not a bundler. The exe produced by this library is just a service runner, not a standalone app.

If you want a standalone bundle (i.e. a single exe), use a bundler like [pkg](https://www.npmjs.com/package/pkg). Alternatively, Node.js 20.0.0 shipped with experimental [single executable application (SEA) support](https://nodejs.org/dist/latest-v20.x/docs/api/single-executable-applications.html#single-executable-applications). Beware that these processes merely merge your code with the node executable, so the file size is large, even for a simple "hello world" script (~60MB for pkg/~90MB for SEA). Electron has some optimization capabilities that can get executables down to ~45MB after unnecessary extras are stripped out.

**Reply by raikm** [reply]
Thank you for the prompt response! 
Just to clarify, does using the service runner in the form of the .exe mean that I won't need to separately install node on the target systems? 

My goal is to minimize dependencies and avoid manual installations on the target systems.
